File: examples/app-yaml-config/sdk/py/15-endpoints-list.py
#!/usr/bin/env python3
"""F05 / story 01 — list_keys() against endpoint.dev.yaml."""
from __future__ import annotations

import json
from pathlib import Path

from app_yaml_fetch_config import EndpointConfigSDK, load_config_from_file

HERE = Path(__file__).resolve().parent
FILE = (HERE / ".." / ".." / ".." / "server" / "config" / "endpoint.dev.yaml").resolve()

load_config_from_file(str(FILE))
sdk = EndpointConfigSDK(file_path=str(FILE))

keys = sorted(sdk.list_keys())
print(json.dumps({"endpoints": keys, "count": len(keys)}, indent=2))

File: examples/app-yaml-config/sdk/mjs/15-endpoints-list.mjs
#!/usr/bin/env node
// F05 / story 01 — listKeys() against endpoint.dev.yaml.
import { EndpointConfigSDK, loadConfigFromFile } from "@ployglot/app-yaml-fetch-config";
import { fileURLToPath } from "node:url";
import { dirname, resolve } from "node:path";

const HERE = dirname(fileURLToPath(import.meta.url));
const FILE = resolve(HERE, "../../../server/config/endpoint.dev.yaml");

loadConfigFromFile(FILE);
const sdk = new EndpointConfigSDK({ filePath: FILE });

const keys = sdk.listKeys().sort();
console.log(JSON.stringify({ endpoints: keys, count: keys.length }, null, 2));
